projects/core/src/error-handling/effects-error-handler/effects-error-handler.service.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpErrorResponse } from '@angular/common/http';
+import { ErrorHandler, Injectable, inject } from '@angular/core';
+import { Action } from '@ngrx/store';
+import { ErrorAction, HttpErrorModel } from '../../model/index';
+import { WindowRef } from '../../window';
+
+@Injectable()
+export class EffectsErrorHandlerService {
+  protected errorHandler: ErrorHandler = inject(ErrorHandler);
+  protected windowRef = inject(WindowRef);
+
+  handleError(action: ErrorAction): void {
+    const error: unknown = action.error;
+
+    // Http errors are already handled in HttpErrorHandlerInterceptor.
+    // To avoid duplicate errors we want to check if the error is not of type
+    // HttpErrorModel or HttpErrorResponse.
+    const isNotHttpError =
+      !(error instanceof HttpErrorModel) &&
+      !(error instanceof HttpErrorResponse);
+
+    if (isNotHttpError && this.shouldHandleError(error)) {
+      this.errorHandler.handleError(error);
+    }
+  }
+
+  /** Here we want to filter which error actions should be handled.
+   * By default, we check if action implements interface ErrorAction  */
+  filterActions(action: Action): action is ErrorAction {
```

Comment:
[MEDIUM]
`filterActions()` vs `isErrorAction()`?

Reply:
That's funny - I have feeling like I've already had such a conversation with Kris Pawelczak and I'm not sure why the name wasn't adjusted. Thanks!